cmd/nri-kubernetes/main.go

```diff
@@ -255,21 +256,23 @@ func buildClients(c *config.Config) (*clusterClients, error) {
 	}, nil
 }
 
-func createIntegrationWithHTTPSink(httpServerPort string) (*integration.Integration, error) {
+func createIntegrationWithHTTPSink(config *config.Config) (*integration.Integration, error) {
 	c := pester.New()
-	c.Backoff = pester.LinearBackoff
-	c.MaxRetries = 5
-	c.Timeout = sink.DefaultRequestTimeout
+	c.Backoff = func(retry int) time.Duration {
+		return config.Sink.HTTP.BackoffDelay
+	}
+	c.MaxRetries = int(config.Sink.HTTP.Timeout / time.Second) // As many attempts as the Context timeout allows
```

Comment:
```suggestion
	// Allow as many attempts as seconds are in the global timeout.
	// This is a rough and conservative approximation as we don't actually need to enforce a number of retries: 
	// We simply rely on the global timeout instead. However, pester requires this value to be set regardless.
	c.MaxRetries = int(config.Sink.HTTP.Timeout / time.Second)
```

Reply:
I hesitated to write such a big comment. Good to know that I should have done it.